.lintr

```diff
@@ -1,7 +1,15 @@
-linters: linters_with_defaults(
-    line_length_linter = line_length_linter(80),
-    cyclocomp_linter = cyclocomp_linter(complexity_limit = 20L),
-    object_usage_linter = NULL)
-exclusions: c(list.files(path = "tests/", recursive = T, full.names = T),
-    "vignettes/germany-age-stratified-nowcasting.Rmd.orig")
-exclude: "# nolint"
+linters: linters_with_tags(
+    tags = NULL, # include all linters
```

Comment:
No, I think you're fine. I personally don't really like most of the disabled linters here because I find them too noisy.

- `implicit_integer_linter` requires using `1L` instead of `1`, which becomes annoying when you have `df[1, ]`
- `extraction_operator_linter` requires using `[[` over `[` whenever possible, which I find less readable
- `undesirable_function_linter` flags `library()` use in vignettes and other files outside of `R/` so too many false positive. It could be reactivated with a subset of functions you want to avoid everywhere though (e.g. `sapply()`
- `function_argument_linter` requires that arguments with defaults appear after arguments without defaults. It may be something to consider here as this is part of the official tidyverse guidance. But I don't always follow this rule myself.
- `function_argument_linter` is still experimental and has too many false positives. It also clashes with your formatting style for argument definition.
- `object_name_linter` flags things like `is.Date()`

Reply:
Okay, all good to know. I hadn't realised arguments with defaults ahead of args without was good practice. We can probably address this on a case-by-case basis but as you say there will likely be some areas where we still want to do this.

> formatting style for argument definition.

By this do you mean having 2nd line (and so on indented to match the first line? I am not particularly wedded to that approach. 

We could potentially consider adding a subset of this info as comments the `.lintr` for future updating ease etc. but not required.

Reply:
> By this do you mean having 2nd line (and so on indented to match the first line? I am not particularly wedded to that approach.

Yes.

An alternative could be:

```r
f <- function(
  arg1,
  arg2
)
```

But as mentioned, even with this, it may be more reasonable to keep `indentation_linter` off for now because many false positives have been reported.

> We could potentially consider adding a subset of this info as comments the .lintr for future updating ease etc. but not required.

lintr documentation is quite good on this aspect so at best it would be redundant and at worst it could become outdated so I would probably not recommend it. But if it's something you'd like to have, I can still add it :shrug:

Reply:
> lintr documentation is quite good on this aspect so at best it would be redundant and at worst it could become outdated so I would probably not recommend it.

No this makes sense. Good point. 

> An alternative could be:

Ah right okay. I do quite like this approach but it can lead to very long files.